adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleNativeAd.m

```diff
@@ -0,0 +1,248 @@
+  VungleAdNetworkExtras *networkExtras = [_adConfiguration extras];
+  if (networkExtras.nativeAdOptionPosition != 0) {
+    _nativeAd.adOptionsPosition = networkExtras.nativeAdOptionPosition;
+  }
+  [_nativeAd loadAd];
+}
+
+#pragma mark - GADMediatedUnifiedNativeAd
+
+- (nullable NSString *)headline {
+    return _nativeAd.title;
+}
+
+- (nullable NSArray<GADNativeAdImage *> *)images {
+  return nil;
+}
+
+- (nullable NSString *)body {
+    return _nativeAd.bodyText;
+}
+
+- (nullable GADNativeAdImage *)icon {
+    return [[GADNativeAdImage alloc] initWithImage:_nativeAd.iconImage];
+}
+
+- (nullable NSString *)callToAction {
+    return _nativeAd.callToAction;
+}
+
+- (nullable NSDecimalNumber *)starRating {
+  return [[NSDecimalNumber alloc] initWithDouble:_nativeAd.adStarRating];
+}
+
+- (nullable NSString *)store {
+  return nil;
+}
+
+- (nullable NSString *)price {
+  return nil;
+}
+
+- (nullable NSString *)advertiser {
+  return _nativeAd.sponsoredText;
+}
+
+- (nullable NSDictionary<NSString *, id> *)extraAssets {
+  return nil;
+}
+
+- (nullable UIView *)adChoicesView {
+  return nil;
+}
+
+- (nullable UIView *)mediaView {
+  return _mediaView;
+}
+
+- (BOOL)hasVideoContent {
+  return YES;
+}
+
+- (void)didRenderInView:(nonnull UIView *)view
+    clickableAssetViews:(nonnull NSDictionary<GADNativeAssetIdentifier, UIView *> *)clickableAssetViews
+ nonclickableAssetViews:(nonnull NSDictionary<GADNativeAssetIdentifier, UIView *> *)nonclickableAssetViews
+         viewController:(nonnull UIViewController *)viewController {
+  NSArray<UIView *> *assets = clickableAssetViews.allValues;
+  UIImageView *iconView = nil;
+  if ([clickableAssetViews[GADNativeIconAsset] isKindOfClass:[UIImageView class]]) {
+    iconView = (UIImageView *)clickableAssetViews[GADNativeIconAsset];
+  }
+    
+  [_nativeAd registerViewForInteraction:view mediaView:_mediaView iconImageView:iconView viewController:viewController clickableViews:assets];
+}
+
+- (void)didUntrackView:(nullable UIView *)view {
+  [_nativeAd unregisterView];
+}
+
+#pragma mark - VungleNativeAdDelegate
+
+- (void)nativeAdDidLoad:(VungleNativeAd *)nativeAd {
+  if (_delegate) {
+    // Already invoked an ad load callback.
+    return;
+  }
+
+  _mediaView = [[VungleMediaView alloc] init];
+  _delegate = _adLoadCompletionHandler(self, nil);
+}
+
+- (void)nativeAd:(VungleNativeAd *)nativeAd didFailWithError:(NSError *)error {
+  _adLoadCompletionHandler(nil, error);
+}
+
+- (void)nativeAdDidClick:(VungleNativeAd *)nativeAd {
+  [_delegate reportClick];
+}
+
+- (void)nativeAdDidTrackImpression:(VungleNativeAd *)nativeAd {
+  [_delegate reportImpression];
+}
+
+#pragma mark - GADMAdapterVungleDelegate
+
+- (void)initialized:(BOOL)isSuccess error:(nullable NSError *)error {
+  if (isSuccess) {
+    [self loadAd];
+  } else {
+    _adLoadCompletionHandler(nil, error);
+  }
+}
+
+- (void)adAvailable {
+    // Do nothing. Native ads utilize a different set of callbacks.
```

Comment:
Nit, prefer:
```
// No-op, Vungle native ads utilize callbacks from VungleNativeAdDelegate.
```

Reply:
I will address this in another commit. It is in internal review currently.